server/modules/analysis/repository/processingStepCalculationRepository.js

```diff
@@ -17,8 +18,61 @@ export const insertCalculationStep = async (surveyId, processingStepUuid, index,
     camelize
   )
 
-// ====== READ
-
 // ====== UPDATE
 
+export const updateCalculationIndex = async (surveyId, processingStepUuid, indexFrom, indexTo, client = db) => {
+  const indexPlaceholder = -1
+  const queries = []
+
+  // set index placeholder for element being edited
+  queries.push(client.one(`
+    UPDATE ${getSurveyDBSchema(surveyId)}.processing_step_calculation
+    SET index = $3
+    WHERE processing_step_uuid = $1
+    AND index = $2
+    RETURNING uuid
+    `,
+    [processingStepUuid, indexFrom, indexPlaceholder]
+  ))
+
+  // decrement previous calculations (element is moved forward)
+  for (let i = indexFrom + 1; i <= indexTo; i++) {
+    queries.push(client.query(`
+      UPDATE ${getSurveyDBSchema(surveyId)}.processing_step_calculation
+      SET index = index - 1
+      WHERE processing_step_uuid = $1
+      AND index = $2
+    `,
+      [processingStepUuid, i]
+    ))
+  }
+
+  // increment next calculations (element is moved backward)
+  for (let i = indexFrom - 1; i >= indexTo; i--) {
+    queries.push(client.query(`
+      UPDATE ${getSurveyDBSchema(surveyId)}.processing_step_calculation
+      SET index = index + 1
+      WHERE processing_step_uuid = $1
+      AND index = $2
+    `,
+      [processingStepUuid, i]
+    ))
+  }
+
+  // set index for element being edited
+  queries.push(client.one(`
+    UPDATE ${getSurveyDBSchema(surveyId)}.processing_step_calculation
```

Comment:
since this query is the same as the first one, can you put it in a constant or move it to an internal function?

Reply:
ok done